src/SFA.DAS.AssessorService.Web/Controllers/Apply/StandardController.cs

```diff
@@ -241,182 +243,200 @@ public async Task<IActionResult> StandardDetails(string referenceNumber)
+            {
+                Id = id,
+                Search = ModelState
+                    .GetAttemptedValueWhenInvalid(nameof(ApplyStandardSearchViewModel.Search), search)
+            };
+
+            return View(viewModel);
         }
 
-        [HttpPost("Standard/{id}")]
-        public async Task<IActionResult> Search(StandardVersionViewModel model)
+        [PrivilegeAuthorize(Privileges.ApplyForStandard)]
+        [HttpPost("standard/{id}/apply-standard", Name = ApplyStandardSearchRoutePost)]
+        [ModelStatePersist(ModelStatePersist.Store)]
+        public IActionResult ApplyStandardSearch(ApplyStandardSearchViewModel model)
         {
-            if (string.IsNullOrEmpty(model.StandardToFind) || model.StandardToFind.Length <= 2)
+            if (model == null)
+                throw new ArgumentException("Value cannot be null or empty", nameof(model));
+
+            if (model.Id == Guid.Empty)
+                throw new ArgumentException($"Value of {nameof(model.Id)} cannot be empty");
+
+            if (!ModelState.IsValid)
             {
-                ModelState.AddModelError(nameof(model.StandardToFind), "Enter a valid search string (more than 2 characters)");
-                TempData["ShowErrors"] = true;
-                return RedirectToAction(nameof(Index), new { model.Id });
+                return RedirectToRoute(ApplyStandardSearchRouteGet, new { id = model.Id, search = model.Search });
             }
 
-            var standards = await _standardVersionApiClient.GetLatestStandardVersions();
-            model.Results = standards
-                .Where(s => s.Title.Contains(model.StandardToFind, StringComparison.InvariantCultureIgnoreCase))
-                .ToList();
-
-            return View("~/Views/Application/Standard/FindStandardResults.cshtml", model);
+            return RedirectToRoute(ApplyStandardSearchResultsRouteGet, new { id = model.Id, search = model.Search });
         }
 
-        [HttpGet("standard/view-standard/{standardReference}")]
-        public async Task<IActionResult> ViewStandard(string standardReference)
+        [PrivilegeAuthorize(Privileges.ApplyForStandard)]
+        [HttpGet("standard/{id}/apply-standard/{search}/results", Name = ApplyStandardSearchResultsRouteGet)]
+        public async Task<IActionResult> ApplyStandardSearchResults(Guid id, string search)
         {
-            var contact = await GetUserContact();
-            var org = await _orgApiClient.GetOrganisationByUserId(contact.Id);
+            if (id == Guid.Empty)
+                throw new ArgumentException($"Value of {nameof(id)} cannot be empty");
 
-            var existingApplications = (await _applicationApiClient.GetStandardApplications(contact.Id))?
-                .Where(p => p.ApplicationStatus != ApplicationStatus.Declined);
+            if (string.IsNullOrEmpty(search))
+                throw new ArgumentException($"Value of {nameof(search)} cannot be null or empty");
```

Comment:
[SonarCloud makes a good point](https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=1526&id=SkillsFundingAgency_das-assessor-service&open=AYlAixI_iks5oidIUBLY&tab=code) about the newly added parameter checks in these async methods. 

The only things I would add is that exceptions being far-removed in time from the actual error would make the logs more confusing, and to suggest using guard clauses in place of this

Reply:
Split methods are per Sonar Cloud suggestions